modules/extensions-support/src/main/java/org/mule/module/extension/internal/runtime/DefaultExecutionMediator.java

```diff
@@ -243,7 +243,109 @@ private void intercept(List<Interceptor> interceptors, Consumer<Interceptor> clo
         });
     }
 
-    private Throwable handleException(Exception e)
+    private class OperationRetryCallBack implements RetryCallback
+    {
+
+        private OperationExecutor operationExecutor;
+        private final OperationContext context;
+        private final List<Interceptor> interceptorList;
+        private OperationExecutionResult operationExecutionResult;
+
+        private OperationRetryCallBack(OperationExecutor operationExecutor, OperationContext context, List<Interceptor> interceptorList)
+        {
+
+            this.operationExecutor = operationExecutor;
+            this.context = context;
+            this.interceptorList = interceptorList;
+        }
+
+        @Override
+        public void doWork(RetryContext retryContext) throws Exception
+        {
+            operationExecutionResult = executeWithInterceptors(operationExecutor, context, interceptorList, new ValueHolder<>());
+
+            if (!operationExecutionResult.isOk())
+            {
+                if (operationExecutionResult.getException() instanceof ConnectionException || (operationExecutionResult.getRetryRequest().isPresent() && operationExecutionResult.getRetryRequest().get().isRetryRequested()))
+                {
+                    throw (Exception) operationExecutionResult.getException();
```

Comment:
I would do a check. If it's an exception then throw it as is. If it's a throwable then wrap it. But do it in a way in which you minimize the amount of instanceof checks. For example, if you already checked that it's a connection exception, then you already know it's an exception.